### ISSUE TITLE
bug(MatSelect): Cannot clear the value of mat-select when [multiple] is specified

### ISSUE THREAD

**Issue by DibyodyutiMondal** [angular/components#22453]
#### Reproduction

Working on a reproduction
 
 
#### Expected Behavior
The documentation at https://material.angular.io/components/select/overview#resetting-the-select-value says

> If you want one of your options to reset the select's value, you can omit specifying its value.

#### Actual Behavior

When the `multiple` attribute is specified, clicking on the option without a specified value inserts `null` or `undefined` into the array of values instead of clearing it

#### Environment

- Angular: 11.2.9
- CDK/Material: 11.2.8
- Browser(s): Edge (but I assume it'll be the same in Chrome, since the bug comes from the component logic, not template/style)
- Operating System (e.g. Windows, macOS, Ubuntu):  Windows

**Comment by mmalerba** [comment]
I don't think this method of resetting the value really makes sense for a multi-select. We should update the docs to clarify that.

**Comment by crisbeto** [comment]
Should be resolved by the `canSelectNullableOptions` input.
